src/diffusers/loaders.py

```diff
@@ -1027,14 +1032,21 @@ def load_lora_into_text_encoder(cls, state_dict, network_alpha, text_encoder, lo
         # then the `state_dict` keys should have `self.unet_name` and/or `self.text_encoder_name` as
         # their prefixes.
         keys = list(state_dict.keys())
-        if all(key.startswith(cls.unet_name) or key.startswith(cls.text_encoder_name) for key in keys):
+        prefix = None
+
+        if any(cls.text_encoder_name in key for key in keys):
             # Load the layers corresponding to text encoder and make necessary adjustments.
-            text_encoder_keys = [k for k in keys if k.startswith(cls.text_encoder_name)]
+            if isinstance(text_encoder, CLIPTextModel):
+                prefix = cls.text_encoder_name
+            elif isinstance(text_encoder, CLIPTextModelWithProjection):
+                prefix = f"{cls.text_encoder_name}_2"
+
```

Comment:
Hmm, I think this check is a bit brittle and will break other pipelines which happen to have a CLIPTextModelWithProjection class for their `text_encoder` key. I don't know if we actively support lora training for any of them (karlo is the only one I can remember off the top of my head), but regardless this is a very implicit contract that is easy to break on accident. 

It would be best if the function instead took the prefix in the state dict as an argument defaulting to `text_encoder`. Then the sdxl pipeline can override `load_lora_weights`. This also gets rid of the additional `is_sdxl` check in the `load_lora_weights` function. 


i.e.

```
class LoraLoaderMixin:
    def load_lora_weights():
        # leave as is

    def load_lora_into_text_encoder(..., prefix='text_encoder'):
        # replace `cls.text_encoder_name` in body with `prefix`
        
class SDXLPipeline:
    def load_lora_weights():
       # load unet ...
        self.load_lora_into_text_encoder(self.text_encoder, 'text_encoder')
        self.load_lora_into_text_encoder(self.text_encoder_2, 'text_encoder')
```

A good rule of thumb is any time we see ourselves looking at the structure of keys or weights in the state dict to implicitly figure out something about the model, ask if we can instead explicitly handle the case in the code and leave the contents of the state dict opaque.